tests/admin_views/tests.py

```diff
@@ -6166,3 +6167,39 @@ def test_implicitly_generated_pk(self):
         post_data = {'name': '2'}
         response = self.client.post(reverse('admin:admin_views_implicitlygeneratedpk_change', args=(1,)), post_data)
         self.assertEqual(response.status_code, 302)
+
+
+@override_settings(
+    PASSWORD_HASHERS=['django.contrib.auth.hashers.SHA1PasswordHasher'],
+    ROOT_URLCONF="admin_views.urls",
+)
+class SiteEachContextTest(TestCase):
+    """
+    Check each_context contains the documented variables
+    """
+    @classmethod
+    def setUpTestData(cls):
+        cls.u1 = User.objects.create(
+            id=100, password='sha1$995a3$6011485ea3834267d719b4c801409b8b1ddd0158',
+            last_login=datetime.datetime(2007, 5, 30, 13, 20, 10), is_superuser=True, username='super',
+            first_name='Super', last_name='User', email='super@example.com',
+            is_staff=True, is_active=True, date_joined=datetime.datetime(2007, 5, 30, 13, 20, 10)
+        )
+
+    def setUp(self):
+        self.factory = RequestFactory()
+
+    def test_each_context(self):
+        request = self.factory.get(reverse('admin:index'))
+        request.user = self.u1
+        ctx = site.each_context(request)
+        self.assertIn('site_header', ctx)
+        self.assertEqual(ctx['site_header'], 'Django administration')
+        self.assertIn('site_title', ctx)
+        self.assertEqual(ctx['site_title'], 'Django site admin')
+        self.assertIn('site_url', ctx)
+        self.assertEqual(ctx['site_url'], '/my-site-url/')
+        self.assertIn('has_permission', ctx)
+        self.assertEqual(ctx['has_permission'], True)
+        self.assertIn('available_apps', ctx)
+        self.assertNotEqual(ctx['available_apps'], [])
```

Comment:
If we are documenting this, I think we should try to add assertions for its contents to prevent breaking it. Also I think the documentation should be more specific about what it contains.